packages/vsce/src/SshConfigUtils.ts

```diff
@@ -370,46 +392,156 @@ export class SshConfigUtils {
         if (selectedConfig?.password) config.secure.push("password" as never);
         if (selectedConfig?.keyPassphrase) config.secure.push("keyPassphrase" as never);
 
+        configApi.profiles.defaultSet("ssh", selectedConfig?.name!);
         configApi.profiles.set(selectedConfig?.name!, config);
         await profInfo.getTeamConfig().save();
     }
 
-    public static async createZoweSchema(): Promise<void> {
+    // Cloned method
+    public static async createZoweSchema(global: boolean): Promise<void> {
         try {
             const user = false;
-            const global = true;
-            const rootPath = FileManagement.getZoweDir();
             const workspaceDir = ZoweVsCodeExtension.workspaceRoot;
 
             const config = await imperative.Config.load("zowe", {
                 homeDir: FileManagement.getZoweDir(),
-                projectDir: FileManagement.getFullPath(rootPath),
+                projectDir: workspaceDir?.uri.fsPath,
             });
-            if (workspaceDir != null) {
-                config.api.layers.activate(user, global, rootPath);
-            }
+
+            config.api.layers.activate(user, global);
 
             const knownCliConfig: imperative.ICommandProfileTypeConfiguration[] = [
                 ZosUssProfile,
                 ProfileConstants.BaseProfile,
             ];
+
+            if (global) knownCliConfig.push(ZosmfProfile, ZosTsoProfile);
+
             config.setSchema(imperative.ConfigSchema.buildSchema(knownCliConfig));
 
             // Note: IConfigBuilderOpts not exported
             // const opts: IConfigBuilderOpts = {
-            const opts: any = {
+            const opts: imperative.IConfigBuilderOpts = {
                 // getSecureValue: this.promptForProp.bind(this),
                 populateProperties: true,
             };
             // Build new config and merge with existing layer
             const impConfig: Partial<imperative.IImperativeConfig> = {
-                profiles: knownCliConfig,
+                profiles: [ProfileConstants.BaseProfile],
                 baseProfile: ProfileConstants.BaseProfile,
             };
             const newConfig: imperative.IConfig = await imperative.ConfigBuilder.build(impConfig, global, opts);
-
             config.api.layers.merge(newConfig);
             await config.save(false);
         } catch (err) {}
     }
+    private static async validateConfig(newConfig: sshConfigExt): Promise<boolean | string> {
+        const attemptConnection = async (config: sshConfigExt): Promise<boolean> => {
```

Comment:
Curious if we should port this function to the `@zowe/zos-uss-for-zowe-sdk` package and perhaps leverage (or deprecate) the `Shell.isConnectionValid` function. 😋